tests/test_server.py

```diff
@@ -149,7 +164,7 @@ def _handler(params):
                         {
                             "code": "F401",
                             "codeDescription": {
-                                "href": "https://docs.astral.sh/ruff/rules/unused-import"
+                                "href": expected_docs_url + "rules/unused-import"
```

Comment:
I wish we could just store "fixtures by version" or something, rather than having to construct these and encode facts like "Which URL is present at which version?" in the test logic. Like, in Rust, we'd just run the command with different versions and snapshot the output, which feels easier to maintain and understand. I don't know if I have a concrete suggestion here -- more reacting to the increased logic in the tests now.

Reply:
Hm interesting yeah.. this whole bit would make more sense as a snapshot too.

A strong case for us building Python test tooling <3